templates/Dockerfile.template

```diff
@@ -0,0 +1,305 @@
+  wget "${base_url}/prometheus_rabbitmq_exporter-3.7.9.1.ez"; \
+  \
+  mkdir -p "/licenses" && cd "/licenses"; \
+  readonly git_base='https://raw.githubusercontent.com/deadtrickster'; \
+  \
+  mkdir accept && wget -O accept/LICENSE "${git_base}/accept/master/LICENSE"; \
+  mkdir prometheus && wget -O prometheus/LICENSE "${git_base}/prometheus.erl/master/LICENSE"; \
+  mkdir prometheus_cowboy && wget -O prometheus_cowboy/LICENSE "${git_base}/prometheus-cowboy/master/LICENSE"; \
+  mkdir prometheus_httpd && wget -O prometheus_httpd/LICENSE "${git_base}/prometheus-httpd/master/LICENSE"; \
+  mkdir prometheus_rabbitmq_exporter && wget -O prometheus_rabbitmq_exporter/LICENSE "${git_base}/prometheus_rabbitmq_exporter/master/LICENSE"; \
+  \
+  sha256sum -c /license-checksums
+
+
+FROM {{ .From }}
+
+RUN set -eux; \
+	apt-get update; \
+	apt-get install -y --no-install-recommends \
+# grab gosu for easy step-down from root
+		gosu \
+	; \
+	rm -rf /var/lib/apt/lists/*; \
+# verify that the "gosu" binary works
+	gosu nobody true
+
+# Default to a PGP keyserver that pgp-happy-eyeballs recognizes, but allow for substitutions locally
+ARG PGP_KEYSERVER=ha.pool.sks-keyservers.net
+# If you are building this image locally and are getting `gpg: keyserver receive failed: No data` errors,
+# run the build with a different PGP_KEYSERVER, e.g. docker build --tag rabbitmq:3.7 --build-arg PGP_KEYSERVER=pgpkeys.eu 3.7/ubuntu
+# For context, see https://github.com/docker-library/official-images/issues/4252
+
+# Using the latest OpenSSL LTS release, with support until September 2023 - https://www.openssl.org/source/
+ENV OPENSSL_VERSION {{ $openssl.Version }}f
+ENV OPENSSL_SOURCE_SHA256="{{ $openssl.Sha256 }}"
+# https://www.openssl.org/community/omc.html
+
+# Use the latest stable Erlang/OTP release (https://github.com/erlang/otp/tags)
+ENV OTP_VERSION {{ $otp.Version }}
+# TODO add PGP checking when the feature will be added to Erlang/OTP's build system
+# http://erlang.org/pipermail/erlang-questions/2019-January/097067.html
+ENV OTP_SOURCE_SHA256="{{ $otp.Sha256 }}"
+
+RUN mkdir ~/.gnupg \
+    && echo "disable-ipv6" >> ~/.gnupg/dirmngr.conf
+
+# Install dependencies required to build Erlang/OTP from source
+# http://erlang.org/doc/installation_guide/INSTALL.html
+# autoconf: Required to configure Erlang/OTP before compiling
+# dpkg-dev: Required to set up host & build type when compiling Erlang/OTP
+# gnupg: Required to verify OpenSSL artefacts
+# libncurses5-dev: Required for Erlang/OTP new shell & observer_cli - https://github.com/zhongwencool/observer_cli
+RUN set -eux; \
+	\
+	savedAptMark="$(apt-mark showmanual)"; \
+	apt-get update; \
+	apt-get install --yes --no-install-recommends \
+		autoconf \
+		ca-certificates \
+        dirmngr \
+		dpkg-dev \
+		gcc \
+		gnupg2 \
+		libncurses5-dev \
+        m4 \
+		make \
+		wget \
+	; \
+	rm -rf /var/lib/apt/lists/*; \
+	\
+	OPENSSL_SOURCE_URL="https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz"; \
```

Comment:
I have concern about installing openssl version this way:
1. It should be updated with application every time in case of security issues. So it means we should track openssl as well.
2. It will generate additional layer in docker image which don't needed by application. 
3. To have gcc and other build tools in container will increase security risks. 

I think in this case we can try to switch main image to debian10 it will be simpler. 
Or we should install openssl from backports.

Also this changes requires to have OSPO review for this container.

Reply:
@vcanaa can you please review carefully again? 

@aav66 I'm assigning an internal bug so you can track this case.